common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapConfigStoreTaskProvider.java

```diff
@@ -164,9 +166,20 @@ protected void findSecurityInstanceId(Link[] objectLinks) {
 
     protected void findServerInstanceId(BootstrapReadResponse readResponse) {
         this.serverInstances = new HashMap<>();
-        ((LwM2mObject) readResponse.getContent()).getInstances().values().forEach(instance -> {
-            serverInstances.put(((Long) instance.getResource(0).getValue()).intValue(), instance.getId());
-        });
+        try {
+            ((LwM2mObject) readResponse.getContent()).getInstances().values().forEach(instance -> {
+                var shId = OPAQUE.equals(instance.getResource(0).getType()) ? new BigInteger((byte[]) instance.getResource(0).getValue()).intValue() : instance.getResource(0).getValue();
+                int shortId;
+                if (shId instanceof Long) {
+                    shortId = ((Long) shId).intValue();
+                } else {
+                    shortId = (int) shId;
+                }
+                serverInstances.put(shortId, instance.getId());
+            });
+        } catch (Exception e) {
+            log.error("", e);
+        }
         if (this.securityInstances != null && this.securityInstances.size() > 0 && this.serverInstances != null && this.serverInstances.size() > 0) {
```

Comment:
Always put something meaningful instead of "". It should help to identify the place where we catched the exception.